### ISSUE TITLE
python3.withPackages example doesn't work in shell.nix

### ISSUE THREAD

**Issue by t184256** [NixOS/nixpkgs#32372]
## Issue description

I follow [example "9.11.1.1.3. Temporary Python environment with nix-shell"](https://nixos.org/nixpkgs/manual/#using-python) and it works.
I follow example "9.11.1.1.3.1. Load environment from .nix expression", and it doesn't.

### Steps to reproduce

Run `nix-shell -p 'python35.withPackages (ps: [ps.numpy ps.toolz])'`, get dropped into a shell with python and numpy.

Save
```{.nix}
with import <nixpkgs> {};

python35.withPackages (ps: [ps.numpy ps.toolz])
```
to `shell.nix` and run nix-shell. Get dropped into a shell without python. It still downloads packages beforehand though.

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.3, NixOS, 17.09.2281.b4a0c011e81 (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(monk): `""`
 - channels(root): `"nixos-17.09.2281.b4a0c011e81"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

and same on unstable.

**Comment by FRidh** [comment]
Last line should be
```
(python35.withPackages (ps: [ps.numpy ps.toolz])).env
```

**Comment by FRidh** [comment]
Fixed on master with 41ff561b7581a6331e214714401be92b63785f60 and 17.09 with d2dff8d0bea.
